roles/get-sitename/tasks/main.yml

```diff
@@ -53,28 +53,66 @@
   when:
     acsf_environment != ""
 
-- name: Get list of sites on Site Factory
-  uri:
-    url: "https://www.{{ sitefactory_environment }}cardinalsites.acsitefactory.com/api/v1/sites?limit=100"
-    method: GET
-    user: "{{ acquia_username }}"
-    password: "{{ acquia_api_key }}"
-    force_basic_auth: yes
-    return_content: yes
-  register: existing_sites
+# - name: Get list of sites on Site Factory
+#   uri:
+#     url: "https://www.{{ sitefactory_environment }}cardinalsites.acsitefactory.com/api/v1/sites?limit=100"
+#     method: GET
+#     user: "{{ acquia_username }}"
+#     password: "{{ acquia_api_key }}"
+#     force_basic_auth: yes
+#     return_content: yes
+#   register: existing_sites
 
-#- name: Display all site names
-#  debug:
-#    var: item
-#  with_items: "{{ existing_sites | json_query('json.sites[*].site') }}"
+- name: Get sites.json from remote
+  command: drush @acsf{{ drush_environment }}.cardinald7.saml php-eval "include 'sites/sites.php'; drupal_json_output(gardens_site_data_load_file());"
+  register: sites_json_raw
 
-- name: Establish numerical id from existing_sites JSON response
+- name: Get sites.json from remote
   set_fact:
-    site_id: "{{ existing_sites | json_query(existing_sites_query) }}"
+    sites_json: "{{ sites_json_raw.stdout|from_json }}"
+
+# - name: debug sites.json
+#   debug:
+#     var: sites_json
+
+#
+# Creates a format that looks like:
+# {
+#     "id": 5001626,
+#     "site": "crceducation",
+#     "url": "crceducation-dev.cardinalsites.stanford.edu"
+# },
+#
+- name: Manipulate sites.json
+  set_fact:
+    sites_formatted: "{{ sites_formatted|default([]) + [ {'url': item.0, 'id': item.1.acsf_site_id, 'site': item.0|regex_replace('((-dev|-stage|-test).cardinalsites.stanford.edu|.(dev-|stage-|test-)cardinalsites.acsitefactory.com|(-dev|-stage|-test).sites.stanford.edu)', '') } ] }}"
```

Comment:
This task could use some code comments to explain what it's doing, because it's pretty opaque to me. I could not divine how `sites_formatted` could be referencing itself to set itself.

https://www.youtube.com/watch?v=67E42LQsU24